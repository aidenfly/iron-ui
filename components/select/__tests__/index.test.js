import React from 'react';
import { render } from '@testing-library/react';
import { VirtualSelect } from '../index';

const options = [];

for (let i = 0; i < 10000; i++) {
  const value = `${i.toString(36)}${i}`;
  options.push(value);
}

describe('VirtualSelect', () => {
  it('render success', () => {
    const wrapper = render(
      <VirtualSelect>
        {options.map((v) => (
          <div key={v}>{v}</div>
        ))}
      </VirtualSelect>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
