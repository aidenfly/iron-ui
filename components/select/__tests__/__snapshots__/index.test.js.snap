// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualSelect render success 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="iron-select-container"
      >
        <span
          class="iron-select-input ant-input-affix-wrapper"
          style="display: none;"
        >
          <input
            class="ant-input"
            placeholder="please select"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <i
              aria-label="icon: up"
              class="anticon anticon-up"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="up"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </i>
          </span>
        </span>
        <span
          class="iron-select-input ant-input-affix-wrapper"
          style="display: inline-block;"
        >
          <input
            class="ant-input"
            placeholder="please select"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <span
              style="display: inline-block;"
            >
              <i
                aria-label="icon: down"
                class="anticon anticon-down"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="iron-select-container"
    >
      <span
        class="iron-select-input ant-input-affix-wrapper"
        style="display: none;"
      >
        <input
          class="ant-input"
          placeholder="please select"
          readonly=""
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <i
            aria-label="icon: up"
            class="anticon anticon-up"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="up"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
              />
            </svg>
          </i>
        </span>
      </span>
      <span
        class="iron-select-input ant-input-affix-wrapper"
        style="display: inline-block;"
      >
        <input
          class="ant-input"
          placeholder="please select"
          readonly=""
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <span
            style="display: inline-block;"
          >
            <i
              aria-label="icon: down"
              class="anticon anticon-down"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
